### PR TITLE
NC-NEWS-11

### PR DESCRIPTION
**Pull request by jonnywb**
add topic query to GET articles request

### CODE REVIEW

**Comment by jonnywb** [comment]
Conflict fixed!

File: controllers/articles.controllers.js [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+
+  const promiseArticles = [selectArticles(topic)];
+
+  if (topic) {
+    promiseArticles.push(checkExists("topics", "slug", topic));
+  }
+
+  Promise.all(promiseArticles)
+    .then((promiseResults) => {
+      res.status(200).send({ articles: promiseResults[0] });
```
Comment (johnhutcheon) [inline controllers/articles.controllers.js:15]: Really nicely done 🏅

File: utils/utils.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const db = require("../db/connection");
+const format = require("pg-format");
+
+exports.checkExists = (table, column, value) => {
+  const queryString = format(`SELECT * FROM %I WHERE %I = $1;`, table, column);
+  return db.query(queryString, [value]).then(({ rows }) => {
+    if (!rows.length) {
+      return Promise.reject({ status: 404, msg: "Not Found" });
```
Comment (johnhutcheon) [inline utils/utils.js:8]: Lovely utils function!

**Review by johnhutcheon — APPROVED** [review]
Really great work here Jonny, well done! 😄

**Pull request merged**